pkg/asyncexec/cmd.go

```diff
@@ -0,0 +1,105 @@
+package asyncexec
+
+import (
+	"bufio"
+	"os/exec"
+)
+
+// New instanciate a new Cmd object.
+func New(cmd *exec.Cmd, buff int) *Cmd {
+	return &Cmd{
+		cmd:      cmd,
+		stopChan: make(chan struct{}),
+		StdoutCh: make(chan []byte, buff),
+		StderrCh: make(chan []byte, buff),
+	}
+}
+
+// Cmd is a wrapper arround exec.Cmd. Mainly used to execute
+// command asynchronously with/or without output stream.
+type Cmd struct {
+	cmd *exec.Cmd
+
+	stopChan chan struct{}
+	StdoutCh chan []byte
+	StderrCh chan []byte
+}
+
+// Run runs the command. if streamOutput is true, it will spin
+// two goroutine responsible of streaming the stdout and stderr
+func (c *Cmd) Run(streamOutput bool) error {
+	if streamOutput {
+		cmdStdoutReader, err := c.cmd.StdoutPipe()
+		if err != nil {
+			return err
+		}
+		stdoutScanner := bufio.NewScanner(cmdStdoutReader)
+
+		cmdStderrReader, err := c.cmd.StderrPipe()
+		if err != nil {
+			return err
+		}
+		stderrScanner := bufio.NewScanner(cmdStderrReader)
+
+		// Goroutine for stdout
+		go func() {
+			defer close(c.StdoutCh)
+			for stdoutScanner.Scan() {
+				bytes := stdoutScanner.Bytes()
+				c.StdoutCh <- bytes
```

Comment:
Will this flush the buffer when it closes?

Reply:
No, it will signal to the consumers that the channels will no longer stream anything